ectoolkits/analysis/disdeg.py

```diff
@@ -0,0 +1,181 @@
+        _dimensions = atomgroup.universe.dimensions
+        if not (np.any(_dimensions) or np.any(cellpar)):
+            raise ValueError("Need cell info!")
+        if np.any(_dimensions):
+            self.cellpar = _dimensions
+        if np.any(cellpar):
+            self.cellpar = cellpar
+        
+        self.terminal_o_idxs = terminal_oxygen_idxs
+        
+        self.bond_cutoff = cutoff
+        self.min_cutoff = min_cutoff
+        
+        self.chosen_elements = chosen_elements
+        
+        _elements = self._ag.atoms.elements
+        if type(chosen_elements) == str:
+            chosen_element = [chosen_elements]
+        sel = np.empty(len(_elements), dtype=bool)
+        sel.fill(0)
+        for element in chosen_elements:
+            sel = (sel) | (_elements == element)
+        self.conf_idxs = np.where(sel)
+        
+        #self.disdeg_settings = {}
+        
+        pass
+    
+    def _prepare(self): 
+        
+        onums = len(self.terminal_o_idxs)
+        nframes = self.n_frames
+        
+        self._disdeg = np.empty([0, onums], dtype=int)
+        #
+    
+    
+    def _single_frame(self):
+        #print(self.cellpar)
+        pos = self._ag.positions
+        iframe = self._ts.frame
+        _idisdeg = count_Oidxs_bonds(pos, 
+                                     self.terminal_o_idxs, 
+                                     self.conf_idxs, 
+                                     self.cellpar,
+                                     max_cutoff=self.bond_cutoff,
+                                     min_cutoff=self.min_cutoff)
+        self._disdeg = np.concatenate((self._disdeg, [_idisdeg]),axis=0)
+    
+    def _conclude(self):
+        
+        #TODO: to vectorzie it
+        # np.vectoriz
+        _arrs = []
+        for arr in self._disdeg:
+            _arrs.append(_count_disdeg(arr))
+        self.disdeg = np.array(_arrs)
+        
+        _arrs = []
+        for arr in self._disdeg:
+            _arrs.append(_count_percent_o(arr))
+        self.percent_o = np.array(_arrs)
+        
+        _arrs = []
+        for arr in self._disdeg:
+            _arrs.append(_count_percent_oh(arr))
+        self.percent_oh = np.array(_arrs)
+        
+        _arrs = []
+        for arr in self._disdeg:
+            _arrs.append(_count_percent_h2o(arr))
+        self.percent_h2o = np.array(_arrs)
+        
+        _arrs = []
+        for arr in self._disdeg:
+            _arrs.append(_count_percent_h3o(arr))
+        self.percent_h3o = np.array(_arrs)
+    
+        self.cumave_disdeg = cumsum_arr(self.disdeg)
+        self.cumave_percent_o = cumsum_arr(self.percent_o)
+        self.cumave_percent_oh = cumsum_arr(self.percent_oh)
+        self.cumave_percent_h2o = cumsum_arr(self.percent_h2o)
+        self.cumave_percent_h3o = cumsum_arr(self.percent_h3o)
+        #print("finished!")
+    
+    
+    def dump_result(self, selects = ["disdeg","percent_o", "percent_oh", "percent_h2o", "percent_h3o"],
```

Comment:
Coud you write the coordination number of terminal o to another file. with shape of (num_frames, terminal_o_idx)
for example:
```
# two frames and 6 terminal oxygen
[
[0, 2, 3, 1 0, 0],
[1, 2, 3, 0,1, 2]
]
```